homeassistant/components/amcrest/camera.py

```diff
@@ -132,7 +135,22 @@ async def async_setup_platform(
 
     name = discovery_info[CONF_NAME]
     device = hass.data[DATA_AMCREST][DEVICES][name]
-    async_add_entities([AmcrestCam(name, device, hass.data[DATA_FFMPEG])], True)
+    entity = AmcrestCam(name, device, hass.data[DATA_FFMPEG])
+
+    # 2021.9.0 introduced unique id's for the camera entity, but these were not
+    # unique for different resolution streams.  If any cameras were configured
+    # with this version, update the old entity with the new unique id.
+    serial_number = await hass.async_add_executor_job(lambda: device.api.serial_number)  # type: ignore[no-any-return]
```

Comment:
I think the `type: ignore` is needed here because of the lambda. Any thoughts on how to get mypy to know what this is would be appreciated.

Reply:
I still have not seen a good solution, and according to google this is a major deficit.

Reply:
The value is striped everywhere else, why is that not needed here?

Reply:
Good catch, thanks. Added the `.strip()` for the serial number.